scout/adapter/mongo/query.py

```diff
@@ -474,63 +474,88 @@ def coordinate_filter(self, query, mongo_query):
 
         return mongo_query
 
-    def sv_coordinate_query(self, query):
-        """Adds genomic coordinated-related filters to the query object
-            This method is called to buid coordinate query for sv variants
-
-        Args:
-            query(dict): a dictionary of query filters specified by the users
-            mongo_query(dict): the query that is going to be submitted to the database
+    def get_position_query(self, chrom: str, start: int, end: int) -> dict:
+        """Helper function that returns a dictionary containing start and stop coordinates.
+        # Query for overlapping intervals. Taking into account these cases:
+        # Case 1
+        # filter                 xxxxxxxxx
+        # Variant           xxxxxxxx
+
+        # Case 2
+        # filter                 xxxxxxxxx
+        # Variant                    xxxxxxxx
+
+        # Case 3
+        # filter                 xxxxxxxxx
+        # Variant                   xx
+
+        # Case 4
+        # filter                 xxxxxxxxx
+        # Variant             xxxxxxxxxxxxxx
+        """
 
-        Returns:
-            coordinate_query(dict): returned object contains coordinate filters for sv variant
+        return {
+            "$or": [
+                # Case chromosome == end_chrom
+                {
+                    "$and": [
+                        {"chromosome": chrom},
+                        {"end_chrom": chrom},
+                        {
+                            "$or": [
+                                # Overlapping cases 1-4 (chromosome == end_chrom)
+                                {"end": {"$gte": start, "$lte": end}},  # Case 1
+                                {"position": {"$gte": start, "$lte": end}},  # Case 2
+                                {
+                                    "$and": [
+                                        {"position": {"$lte": start}},
+                                        {"end": {"$gte": end}},
+                                    ]
+                                },  # Case 3
+                                {
+                                    "$and": [
+                                        {"position": {"$gte": start}},
+                                        {"end": {"$lte": end}},
+                                    ]
+                                },  # Case 4
+                            ]
+                        },
+                    ]
+                },
+                # Case chromosome != end_chrom, position matching 'chromosome'
```

Comment:
New

Reply:
Nice! Very clear, and while it may perhaps be possible to reduce it to something more compact, as long as it is not really slow why not just keep it readable! 💯